src/qinfer/distributions.py

```diff
@@ -321,6 +322,76 @@ def sample(self, n=1):
         ), len(cumsum_weights) - 1)]
 
     ## MOMENT FUNCTIONS ##
+    @staticmethod
+    def particle_mean(weights, locations):
+        r"""
+        Returns the mean of a function :math:`f` over model
```

Comment:
This is a tough point. It really depends on what `est_mean_fn` is actually needed for. So far, I only spotted usages with the identity function, i.e. giving the mean.

The problem is that for the use case Chris and I had in my mind (circular quantities as shown in the notebook), `particle_mean` and `est_fn` work completely different and one cannot be expressed as the other.